src/Decode.fs

```diff
@@ -1,59 +1,45 @@
 namespace Thoth.Json.Net
 
+open System
+open System.Text.Json
+open System.Globalization
+
 [<RequireQualifiedAccess>]
 module Decode =
-
-    open System.Globalization
-    open Newtonsoft.Json
-    open Newtonsoft.Json.Linq
-    open System.IO
-
     module Helpers =
-        let anyToString (token: JsonValue) : string =
-            if isNull token then "null"
-            else
-                use stream = new StringWriter(NewLine = "\n")
-                use jsonWriter = new JsonTextWriter(
-                                        stream,
-                                        Formatting = Formatting.Indented,
-                                        Indentation = 4 )
-
-                token.WriteTo(jsonWriter)
-                stream.ToString()
-
-        let inline getField (fieldName: string) (token: JsonValue) = token.Item(fieldName)
-        let inline isBool (token: JsonValue) = not(isNull token) && token.Type = JTokenType.Boolean
-        let inline isNumber (token: JsonValue) = not(isNull token) && (token.Type = JTokenType.Float || token.Type = JTokenType.Integer)
-        let inline isIntegralValue (token: JsonValue) = not(isNull token) && (token.Type = JTokenType.Integer)
-        let inline isInteger (token: JsonValue) = not(isNull token) && (token.Type = JTokenType.Integer)
-        let inline isString (token: JsonValue) = not(isNull token) && token.Type = JTokenType.String
-        let inline isGuid (token: JsonValue) = not(isNull token) && token.Type = JTokenType.Guid
-        let inline isDate (token: JsonValue) = not(isNull token) && token.Type = JTokenType.Date
-        let inline isArray (token: JsonValue) = not(isNull token) && token.Type = JTokenType.Array
-        let inline isObject (token: JsonValue) = not(isNull token) && token.Type = JTokenType.Object
-        let inline isUndefined (token: JsonValue) = isNull token
-        let inline isNullValue (token: JsonValue) = isNull token || token.Type = JTokenType.Null
-        let inline asBool (token: JsonValue): bool = token.Value<bool>()
-        let inline asInt (token: JsonValue): int = token.Value<int>()
-        let inline asFloat (token: JsonValue): float = token.Value<float>()
-        let inline asFloat32 (token: JsonValue): float32 = token.Value<float32>()
-        let inline asDecimal (token: JsonValue): System.Decimal = token.Value<System.Decimal>()
-        let inline asString (token: JsonValue): string = token.Value<string>()
-        let inline asArray (token: JsonValue): JsonValue[] = token.Value<JArray>().Values() |> Seq.toArray
+        let anyToString (token: JsonElement) : string =
```

Comment:
most of the root-level changes were in this `Helpers` module, where we interact with the "raw" `JsonElement` AST tokens